### PR TITLE
add rector phpstan and ecs

### PR DESCRIPTION
**Pull request by jayesh-parmar**

### CODE REVIEW

**Comment by jayesh-parmar** [comment]
👍

**Comment by dineshram7660** [comment]
please check all details how to run and setup rector

**Comment by jayesh-parmar** [comment]
👍

File: rector.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        __DIR__ . '/public',
+        __DIR__ . '/resources',
+        __DIR__ . '/routes',
+    ]);
+
+    $rectorConfig->sets([
+        SetList::DEAD_CODE,
+        SetList::CODE_QUALITY,
+        LevelSetList::UP_TO_PHP_82,
+    ]);
```
Comment (hirenkeraliya) [inline rector.php]: Add more sets.

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

File: tests/CreatesApplication.php [hunk 2]
```diff
@@ -1,21 +0,0 @@
-<?php
-
-namespace Tests;
-
-use Illuminate\Contracts\Console\Kernel;
-use Illuminate\Foundation\Application;
-
-trait CreatesApplication
-{
-    /**
```
Comment (dineshram7660) [inline tests/CreatesApplication.php:10]: a kem delete karu che?

**Review by dineshram7660 — CHANGES_REQUESTED** [review]

**Review by hirenkeraliya — COMMENTED** [review]
https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md

Check this list of sets.

File: rector.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        SetList::PHP_56,
+        SetList::PRIVATIZATION,
+        SetList::TYPE_DECLARATION,
+        SetList::EARLY_RETURN,
+        SetList::INSTANCEOF,
+        SetList::GMAGICK_TO_IMAGICK,
+        SetList::CODING_STYLE,
+        SetList::STRICT_BOOLEANS,
+        PHPUnitSetList::PHPUNIT_100,
+        PHPUnitSetList::PHPUNIT_CODE_QUALITY,
```
Comment (hirenkeraliya) [inline rector.php]: Still Missing
---
PSR_4
LARAVEL_ARRAY_STR_FUNCTION_TO_STATIC_CALL
LARAVEL_CODE_QUALITY
LARAVEL_LEGACY_FACTORIES_TO_CLASSES
ARRAY_STR_FUNCTIONS_TO_STATIC_CALL

Not required
---
PHP_70,
PHP_71,
PHP_72,
PHP_73,
PHP_74,
PHP_80,
PHP_81,
PHP_52,
PHP_53,
PHP_54,
PHP_55,
PHP_56,

Duplicate
---
PHP_82,

**Review by hirenkeraliya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**